src/components/Header/index.jsx

```diff
@@ -1,102 +1,120 @@
-import React from 'react';
+import React, { useEffect } from 'react';
 import PropTypes from 'prop-types';
-import { Link } from 'react-router-dom';
-import { Button, Icon, Dropdown } from '@edx/paragon';
+import {
+  Dropdown, Navbar, AvatarButton, Nav,
+} from '@edx/paragon';
 
+import {
+  getAuthenticatedUser, hydrateAuthenticatedUser, getLogoutRedirectUrl,
+} from '@edx/frontend-platform/auth';
 import SidebarToggle from '../../containers/SidebarToggle';
 import Img from '../Img';
 
-import apiClient from '../../data/apiClient';
 import { configuration } from '../../config';
 
 import './Header.scss';
 
-class Header extends React.Component {
-  componentDidMount() {
-    if (this.props.username) {
-      this.props.fetchUserAccount(this.props.username);
-    }
-  }
+export const Logo = ({ enterpriseLogo, enterpriseName }) => {
+  const logo = configuration.LOGO_URL;
+
+  return (
+    <Img
+      src={enterpriseLogo || logo}
+      alt={`${enterpriseName || 'edX'} logo`}
+      onError={(e) => { e.target.src = logo; }}
+    />
+  );
+};
+
+Logo.defaultProps = {
+  enterpriseLogo: configuration.LOGO_URL,
+};
+
+Logo.propTypes = {
+  enterpriseLogo: PropTypes.string,
+  enterpriseName: PropTypes.string.isRequired,
+};
 
-  getProfileIconElement() {
-    const { email, userProfileImageUrl } = this.props;
-    const screenReaderText = `Profile image for ${email}`;
+export const HeaderDropdown = ({ user }) => {
+  const { profileImage, username } = user;
+  const avatarImage = profileImage?.hasImage ? profileImage.imageUrlMedium : null;
+  const avatarScreenReaderText = `Profile image for ${username}`;
 
-    if (userProfileImageUrl) {
-      return <Img className="user-profile-img mr-2" src={userProfileImageUrl} alt={screenReaderText} />;
+  return (
+    <Dropdown>
+      <Dropdown.Toggle
+        as={AvatarButton}
+        src={avatarImage}
+        id="avatar-dropdown"
+        size="md"
+        alt={avatarScreenReaderText}
+      >
+        {username}
+      </Dropdown.Toggle>
+      <Dropdown.Menu>
+        <Dropdown.Item
+          href={getLogoutRedirectUrl()}
+        >
+          Logout
+        </Dropdown.Item>
+      </Dropdown.Menu>
+    </Dropdown>
+  );
+};
+
+HeaderDropdown.propTypes = {
+  user: PropTypes.shape({
+    profileImage: PropTypes.shape({
+      hasImage: PropTypes.bool.isRequired,
+      imageUrlMedium: PropTypes.string,
+    }),
+    username: PropTypes.string.isRequired,
+  }).isRequired,
+};
+
+const Header = ({
+  hasSidebarToggle, enterpriseName, enterpriseLogo,
+}) => {
+  const user = getAuthenticatedUser();
+  useEffect(() => {
+    // if there is no email, the user data has not been hydrated
+    if (user && !user.email) {
+      hydrateAuthenticatedUser();
```

Comment:
dont know how hydrateAuthentication works exactly, there is hopefully not going to be a re execution of this useEffect once this is called?

Reply:
I guess even if it does , this check means it won't try to rehydrate.. so might be okay